frontend/src/components/bucket/BucketPermissionUser.vue

```diff
@@ -0,0 +1,109 @@
+<script setup lang="ts">
+import { onMounted, ref } from 'vue';
+import { storeToRefs } from 'pinia';
+import Button from 'primevue/button';
+import Dropdown from 'primevue/dropdown';
+import { useBucketStore, useUserStore } from '@/store';
+import type { User } from '@/interfaces';
+
+const userStore = useUserStore();
+
+const { permissions } = storeToRefs(useBucketStore());
+const { userSearch } = storeToRefs(useUserStore());
+
+onMounted(() => {
+  reset();
+});
+
+const isAddDisabled: any = ref(true);
+const showUserExistError: any = ref(false);
+const selectedUser = ref();
+
+const emit = defineEmits(['cancel-bucket-permission-user']);
+
+const cancelBucketPermissionUser = async () => {
+  isAddDisabled.value = true;
+  reset();
+  emit('cancel-bucket-permission-user');
+};
+
+const ifUserExist = () => {
+  return permissions.value.some(e => e.fullName === selectedUser.value.fullName);
+};
+
+const onInput = async (event: any) => {
+  showUserExistError.value = ifUserExist();
+  var inputValue = event.target.value;
+  if(inputValue.length > 0) {
+    await userStore.searchUsers({ search: inputValue });
+  }
+  if(inputValue.length < 1) {
+    reset();
+  }
+  if (selectedUser.value !== null && selectedUser.value.fullName === undefined) {
+    isAddDisabled.value = true;
+  }
+};
+
+const onChange = () => {
+  showUserExistError.value = ifUserExist();
+  isAddDisabled.value = (selectedUser.value === null || ifUserExist());
+};
+
+const onAdd = () => {
+  if(selectedUser.value !== null && !ifUserExist()) {
+    permissions.value.push(
+      {
+        userId: selectedUser.value.userId,
+        elevatedRights: true,
```

Comment:
maybe somethingf ro a later PR (?) i think we need to get elevatedRights value by checking the config for their idp.
something like `config.value.idpList.find((idp: any) => idp.idp === userSearch.value[0].idp)?.elevatedRights ?? false;`